### PR TITLE
Cleanup in WagtailAPI module

### LINKED ISSUE #1671: Cachebusting query parameter (e.g. _=1441835249458) not ignored by api
From the [documentation for jQuery.ajax, under "cache"](http://api.jquery.com/jquery.ajax/):
> Setting cache to false will only work correctly with HEAD and GET requests. It works by appending "_={timestamp}" to the GET parameters. The parameter is not needed for other types of requests, except in IE8 when a POST is made to a URL that has already been requested by a GET.

It seems like it's standard practice to ignore the underscore keyword. Unless I'm mistaken this is an oversight and not a disagreement on the principle of the thing.

Reproduce: make an Ajax call to any wagtail API endpoint with the cache flag set to false. Or just navigate to something like `http://localhost:8000/api/v1/pages/?type=home.HomePage&_=1441835249458`

You'll get this message: 
```
{
    "message": "query parameter is not an operation or a recognised field: _"
}
```

### PR DESCRIPTION
**Pull request by kaedroho**
See commit log

### CODE REVIEW

**Comment by kaedroho** [comment]
I'll split this up into 4 PRs
 - Cleanup
 - Replacing custom renderer with ``WagtailAPIRouter``
 - Allow "_" GET parameter
 - DRF browsable API

**Comment by kaedroho** [comment]
Split up. Couldn't deserialise the changes much though

File: wagtail/contrib/wagtailapi/endpoints.py [hunk 1]
```diff
@@ -28,8 +28,8 @@ class BaseAPIEndpoint(GenericViewSet):
     renderer_classes = [WagtailJSONRenderer]
     pagination_class = WagtailPagination
     base_serializer_class = BaseSerializer
-    filter_classes = []
-    queryset = None  # Set on subclasses or implement `get_queryset()`.
+    filter_backends = []
+    model = None # Set on subclass
```
Comment (kaedroho) [inline wagtail/contrib/wagtailapi/endpoints.py]: Need two spaces here

**Comment by SalahAdDin** [comment]
What's about django-rest-framework?

**Pull request merged**